### DISCUSSION TITLE
Watch for input file as webpack\postcss plugin

### DISCUSSION THREAD

**Discussion by blackswanny** [tailwindlabs/tailwindcss#13631] [Help] [answered]
I have tailwind running in two modes
1) As CLI
`tailwindcss --input ./index.css --config tailwind.config.js`
where I also have input file

2) As webpack\postcss plugin ( see below config in webpack.config)

```
{
        test: /\.css$/i,
        use: [
          "style-loader",
          "css-loader",
          {
            loader: require.resolve("postcss-loader"),
            options: {
              postcssOptions: {
                plugins: [
                  require("tailwindcss")(
                    path.resolve("tailwind.config.js")
                  )
                ]
              },
            }
          }
        ]
      },
```
First approach is very slow as it parses all source code

While second approach has issue when I run it as part of webpack dev server, 
it doesn't watch for changes in input file index.css, so to see it, i need to re-run 1)
I searched everywhere and didn't find how to  add input file to plugin configuration, config file tailwind.config.js doesn't accept it

I don't want to run tailwind CLI with "-watch" as well, cause it will require to run two processes - tailwind and webpack in different terminals. Also I tried it, and it takes very long as well, probably, because it doesn't just re-compile input file, but also parses all source code again

**Comment by wongjn** [comment, marked as answer]
For the first approach, consider checking if you have [overly broad `content` patterns](https://tailwindcss.com/docs/content-configuration#pattern-recommendations).

For the second approach, consider checking you have `import`ed the `index.css` into Webpack module graph.

**Comment by blackswanny** [comment]
Thanks for response. I will check how to optimize first approach, but second is my preference
You were right, my input index.css was not part of webpack bundle, so it wasn't watched, once I added it, postcss\tailwind plugin started to quickly update styles with hot-reload.

However, i included not input index.css, but output styles.css of the CLI into webpack before.
The issue is output contains more stuff than input, it includes full complied styles I need, after tailwind adds more, such as medias for screen breakpoints, some dark theme additions and so on
